src/test/java/com/es/phoneshop/DAO/CustomProductDaoTest.java

```diff
@@ -0,0 +1,100 @@
+package com.es.phoneshop.DAO;
+
+import com.es.phoneshop.exception.ProductDefinitionException;
+import com.es.phoneshop.model.entity.Product;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.math.BigDecimal;
+import java.util.Currency;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CustomProductDaoTest {
+    private final Currency usd = Currency.getInstance("USD");
+    private ProductDao productDao;
+    private Product product;
+    private List<Product> productList;
+
+    @Before
+    public void setup() {
+        productDao = CustomProductDao.getInstance();
+
+        product = new Product("test", "test", new BigDecimal(100), usd, 10, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+
+        productList = CustomProductDao.getInstance().getProducts();
+        productList.add(product);//as it's a reference we can add products this way to productDao, i suppose, without using save() and other method from test object
```

Comment:
actually when writing a test you should not rely on real DAO resources
you should mock them
here it's easy to do: create a list of mocked Product-s and via setter add them to dao
it's not a good idea to create tests on real data